diff --git a/userpoints_collect.admin.inc b/userpoints_collect.admin.inc
new file mode 100644
index 0000000..8801518
--- /dev/null
+++ b/userpoints_collect.admin.inc
@@ -0,0 +1,21 @@
+<?php
+
+/**
+ * @file
+ * Provides administrative settings for userpoints collect module.
+ */
+
+
+/**
+ * Administrative Settings Form.
+ */
+function userpoints_collect_admin_settings($form, &$form_state) {
+  $form['userpoints_collect_test'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Test Setting'),
+    '#description' => t('Testing settings for the userpoints collect module'),
+    '#default_value' => variable_get('userpoints_collect_test', ''),
+  );
+
+  return system_settings_form($form);
+}
\ No newline at end of file
diff --git a/userpoints_collect.module b/userpoints_collect.module
index b77931a..b17c63f 100644
--- a/userpoints_collect.module
+++ b/userpoints_collect.module
@@ -33,6 +33,25 @@ function userpoints_collect_theme() {
 }
 
 /**
+ * Implements hook_menu().
+ */
+function userpoints_collect_menu() {
+  return array(
+    'admin/config/people/userpoints/collect' => array(
+      'title' => 'Collect settings',
+      'description' => 'Settings for userpoints collect',
+      'page callback' => 'drupal_get_form',
+      'page arguments' => array('userpoints_collect_admin_settings'),
+      'access callback' => 'userpoints_admin_access',
+      'access arguments' => array('administer'),
+      'file' => 'userpoints_collect.admin.inc',
+      'type' => MENU_LOCAL_TASK,
+      'weight' => 11,
+    ),
+  );
+}
+
+/**
  * Implements hook_block_info().
  */
 function userpoints_collect_block_info() {
diff --git a/userpoints_collect.variable.inc b/userpoints_collect.variable.inc
new file mode 100644
index 0000000..3223d61
--- /dev/null
+++ b/userpoints_collect.variable.inc
@@ -0,0 +1,35 @@
+<?php
+
+/**
+ * @file
+ * Provides variable definitions for userpoints_collect module.
+ */
+
+
+/**
+ * Implements hook_variable_info().
+ */
+function userpoints_collect_variable_info($options) {
+  $variables['userpoints_collect_test'] = array(
+    'type' => 'string',
+    'title' => t('Userpoints Collect Test', array(), $options),
+    'default' => 'Test',
+    'description' => t('Provides a test link for userpoints_collect', array(), $options),
+    'required' => TRUE,
+    'group' => 'userpoints_collect',
+  );
+  return $variables;
+}
+
+/**
+ * Implements hook_variable_group_info().
+ */
+function userpoints_collect_variable_group_info() {
+  $groups['userpoints_collect'] = array(
+    'title' => t('Userpoints Collect'),
+    'description' => t('Userpoints Collect settings.'),
+    'access' => 'administer site configuration',
+    'path' => array('admin/config/people/userpoints/collect'),
+  );
+  return $groups;
+}
